test/federation.js

```diff
@@ -688,3 +688,112 @@ test('subscription server sends update to subscriptions', t => {
     })
   })
 })
+
+test('federation supports loader for __resolveReference function', async (t) => {
+  const app = Fastify()
+  const users = {
+    1: {
+      id: 1,
+      name: 'John',
+      username: '@john'
+    },
+    2: {
+      id: 2,
+      name: 'Jane',
+      username: '@jane'
+    }
+  }
+  const schema = `
+    extend type Query {
+      me: User
+    }
+
+    type User @key(fields: "id") {
+      id: ID!
+      name: String
+      username: String
+    }
+  `
+
+  const resolvers = {
+    Query: {
+      me: () => {
+        return users['1']
+      }
+    }
+  }
+
+  const loaders = {
+    User: {
+      __resolveReference: {
+        async loader (queries, { reply }) {
+          t.deepEqual(queries, [{
+            obj: {
+              __typename: 'User',
+              id: '1'
+            },
+            params: {}
+          }, {
+            obj: {
+              __typename: 'User',
+              id: '2'
+            },
+            params: {}
+          }])
+          return queries.map(({ obj }) => users[obj.id])
+        },
```

Comment:
This needs to be documented / we should add an example somewhere. Note that without loader this is going to create _1+N_ performance issues.

Reply:
Added a new section to the readme